login-tool/tool/src/java/org/sakaiproject/login/tool/SkinnableLogin.java

```diff
@@ -422,13 +422,19 @@ public LoginRenderContext startPageContext(String skin, HttpServletRequest reque
 		String localCSS = serverConfigurationService.getString("local.css", null);
 		String uiService = serverConfigurationService.getString("ui.service", "Sakai");
 		String passwordResetUrl = getPasswordResetUrl();
+		String logInOutUrl = Web.serverUrl(request) + org.sakaiproject.component.cover.ServerConfigurationService.getString("portalPath") + "/login";
```

Comment:
There is already a member variable of `serverConfigurationService` which should have a copy of the service you can use.